Source/Tools/CodeSizeValidation/Main.cpp

```diff
@@ -0,0 +1,432 @@
+#include "DummyHandlers.h"
+#include "FEXCore/Core/Context.h"
+#include "FEXCore/Debug/InternalThreadState.h"
+#include <FEXCore/Config/Config.h>
+#include <FEXCore/Utils/Allocator.h>
+#include <FEXCore/Utils/File.h>
+#include <FEXCore/Utils/FileLoading.h>
+#include <FEXCore/Utils/LogManager.h>
+
+namespace CodeSize {
+  class CodeSizeValidation final {
+    public:
+      struct InstructionStats {
+        uint64_t GuestCodeInstructions{};
+        uint64_t HostCodeInstructions{};
+
+        uint64_t HeaderSize;
+        uint64_t TailSize;
+      };
+
+      using CodeLines = fextl::vector<fextl::string>;
+      using InstructionData = std::pair<InstructionStats, CodeLines>;
+
+      bool ParseMessage(char const *Message);
+
+      InstructionData *GetDataForRIP(uint64_t RIP) {
+        return &RIPToStats[RIP];
+      }
+
+      bool InfoPrintingDisabled() const {
+        return SetupInfoDisabled;
+      }
+
+      void CalculateBaseStats(FEXCore::Context::Context *CTX, FEXCore::Core::InternalThreadState *Thread);
+    private:
+      void ClearStats() {
+        RIPToStats.clear();
+      }
+
+      void SetBaseStats(InstructionStats const &NewBase) {
+        BaseStats = NewBase;
+      }
+
+      void CalculateDifferenceBetweenStats(InstructionData *Nop, InstructionData *Fence);
+
+      uint64_t CurrentRIPParse{};
+      bool ConsumingDisassembly{};
+      InstructionData *CurrentStats{};
+      InstructionStats BaseStats{};
+
+      fextl::unordered_map<uint64_t, InstructionData> RIPToStats;
+      bool SetupInfoDisabled{};
+  };
+
+  const fextl::string RIPMessage = "RIP: 0x";
```

Comment:
Seems like these strings could be constexpr `std::string_view` instances.

Reply:
switched to constexpr string_view